java/src/jmri/PowerManager.java

```diff
@@ -42,20 +42,67 @@ public interface PowerManager extends PropertyChangeProvider {
     static final int ON = 0x02;
     static final int OFF = 0x04;
     static final int IDLE = 0x08; // not supported by some connection types
-
+    
     /**
      * {@link java.beans.PropertyChangeEvent}s are fired with this property name.
      * <p>
      * {@value #POWER}
      */
     static final String POWER = "power"; // as recommended in JavaBeans Spec // NOI18N
+    static final String PROGPOWER = "progpower"; // as recommended in JavaBeans Spec // NOI18N
 
     public void setPower(int v) throws JmriException;
 
     @CheckReturnValue
     public int getPower();
 
-    // to free resources when no longer used
+    /**
+     * Can programming track power be controlled by this interface?
+     * Only some connections supports it.
+     * 
+     * Control may also depend on other system operating modes or state.
+     * 
+     * @return true if track power can be controlled, false otherwise
+     */
+    public default boolean isProgTrackPowerSupported() {
+        return false;
+    }
+ 
+    /**
+     * Set the programming track power state
+     * 
+     * This method must be overridden with system specific behaviour if independent 
+     * programming track power control is possible.
+     * 
+     * @param v the power state
+     * @throws JmriException if not supported or not overridden
```

Comment:
A second `@throws` is needed since this method also throws `UnsupportedOperationException`. The same applies to the `getProgTrackPower()` method.

Reply:
Happy to do this, but there are no CI errors related to this and the IDE does not add the second throw when I create the Javadoc by typing /** followed by return in the line above the method.

Reply:
`UnsupportedOperationException` inherits `RuntimeException` and RuntimeException is an unchecked exception in Java. That means that you can throw an UnsupportedOperationException without declaring it. So stricktly speaking, you don't need a `@throws` javadoc for UnsupportedOperationException, but it's good to have since it explains why and when that exception is thrown.

Reply:
Should the method actually be

`    public default int getProgTrackPower() throws JmriException, UnsupportedOperationException {`

for clarity?

Reply:
Yes, that's good.